kernel/src/transaction/create_table.rs

```diff
@@ -0,0 +1,991 @@
+            }),
+            DataLayout::Clustered(cols) => {
+                // Validate clustering columns
+                Self::validate_clustering_columns(cols, schema)?;
+
+                // Create clustering domain metadata
+                // When column mapping is enabled, we need to store physical column names
+                let clustering_metadata =
+                    ClusteringMetadataDomain::new_with_schema(cols, schema, column_mapping_mode)?;
+                let domain_metadata = clustering_metadata.to_domain_metadata()?;
+
+                // Clustering requires these features
+                let additional_features = vec![
+                    TableFeature::ClusteredTable.to_string(),
+                    TableFeature::DomainMetadata.to_string(),
+                ];
+
+                Ok(ProcessedDataLayout {
+                    partition_columns: vec![],
+                    clustering_domain_metadata: Some(domain_metadata),
+                    additional_writer_features: additional_features,
+                })
+            }
+        }
+    }
+
+    /// Builds a [`Transaction`] that can be committed to create the table.
+    ///
+    /// This method performs validation:
+    /// - Checks that the table path is valid
+    /// - Verifies the table doesn't already exist
+    /// - Validates the schema is non-empty
+    /// - For clustered tables: validates clustering columns exist and adds required features
+    ///
+    /// # Arguments
+    ///
+    /// * `engine` - The engine instance to use for validation
+    /// * `committer` - The committer to use for the transaction
+    ///
+    /// # Errors
+    ///
+    /// Returns an error if:
+    /// - The table path is invalid
+    /// - A table already exists at the given path
+    /// - The schema is empty
+    /// - Clustering columns don't exist in schema or exceed the limit (4)
+    ///
+    /// # Example
+    ///
+    /// ```rust,no_run
+    /// # use delta_kernel::table_manager::TableManager;
+    /// # use delta_kernel::schema::{StructType, DataType, StructField};
+    /// # use delta_kernel::committer::FileSystemCommitter;
+    /// # use std::sync::Arc;
+    /// # use delta_kernel::Engine;
+    /// # fn example(engine: &dyn Engine) -> delta_kernel::DeltaResult<()> {
+    /// # let schema = Arc::new(StructType::try_new(vec![StructField::new("id", DataType::INTEGER, false)]).unwrap());
+    /// let txn = TableManager::create_table("/path/to/table", schema, "MyApp/1.0")
+    ///     .build(engine, Box::new(FileSystemCommitter::new()))?;
+    ///
+    /// // Commit the transaction to create the table
+    /// let result = txn.commit(engine)?;
+    /// # Ok(())
+    /// # }
+    /// ```
+    pub fn build(
+        self,
+        engine: &dyn Engine,
+        committer: Box<dyn Committer>,
+    ) -> DeltaResult<Transaction> {
+        // Validate path
+        let table_url = try_parse_uri(&self.path)?;
+
+        // Validate schema is non-empty
```

Comment:
I think we need more robust schema validation:
- No duplicate column names (case-insensitive check across all nested fields)
- Supported inner types for complex types
- I'm sure kernel java has a ton of additional checks here, too
- Valid column names (if column mapping is NOT enabled). I think when cm is enabled it still cannot have `\n`